src/freenet/clients/http/DiagnosticToadlet.java

```diff
@@ -409,15 +414,60 @@ public int compare(PeerNodeStatus firstNode, PeerNodeStatus secondNode) {
 		}
 		textBuilder.append("\n");
 
-		// drawThreadPriorityStatsBox
-		textBuilder.append("Threads:\n");
-		int[] activeThreadsByPriority = stats.getActiveThreadsByPriority();
-		int[] waitingThreadsByPriority = stats.getWaitingThreadsByPriority();
-		for(int i=0; i<activeThreadsByPriority.length; i++) {
-			textBuilder.append(l10n("running")).append(": ").append(String.valueOf(activeThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-			textBuilder.append(l10n("waiting")).append(": ").append(String.valueOf(waitingThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-		}
-		textBuilder.append("\n");
+			// drawThreadPriorityStatsBox
+			textBuilder.append("Threads:\n");
+			ThreadMXBean threadMxBean = ManagementFactory.getThreadMXBean();
+			Map<Long, Long> threadCPU = new HashMap<Long, Long>();
+			for (ThreadInfo info : threadMxBean.dumpAllThreads(false, false)) {
+			    threadMxBean.getThreadInfo(info.getThreadId());
+				threadCPU.put(info.getThreadId(), threadMxBean.getThreadCpuTime(info.getThreadId()));
+			}
+
+			RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
+			long initialUptime = runtimeMxBean.getUptime();
+
+			try {Thread.sleep(1000);} catch (InterruptedException e) {}
+
+			for (ThreadInfo info : threadMxBean.dumpAllThreads(false, false)) {
+				Long prev = threadCPU.get(info.getThreadId());
+				if (prev == null)
+					continue;
+				threadCPU.put(info.getThreadId(), threadMxBean.getThreadCpuTime(info.getThreadId()) - prev);
+			}
+
+			Float totalCPU = (runtimeMxBean.getUptime() - initialUptime) * 1000000F * 1;
+
+			// ID, Name, Priority, Group (system, main), Status, % CPU
+			textBuilder.append(
+				String.format(
+					"%5s %-30s %5s %10s %-20s %-5s%n",
+					"ID",
+					"Name",
+					"Prio.",
+					"Group",
+					"Status",
+					"% CPU"
+				)
+			);
+
+			for (Thread t : stats.getThreads()) {
+				if (t == null)
+					continue;
+
+				Float cpuUsage = threadCPU.getOrDefault(t.getId(), 0l) / totalCPU;
```

Comment:
Same as above, no reason to box this to a `Float` (and again no reason for `float` at all).